apps/dcos_l4lb/src/dcos_l4lb_mgr.erl

```diff
@@ -181,63 +184,111 @@ start_reconcile_timer() ->
 %%%===================================================================
 
 -spec(handle_reconcile([{key(), [backend()]}], state()) -> state()).
-handle_reconcile(VIPs, #state{route_mgr=RouteMgr, ipvs_mgr=IPVSMgr,
-        tree=Tree, nodes=Nodes, namespaces=Namespaces}=State) ->
+handle_reconcile(VIPs, #state{tree=Tree, nodes=Nodes, namespaces=Namespaces,
+        route_mgr=RouteMgr, ipvs_mgr=IPVSMgr, ipset_mgr=IPSetMgr}=State) ->
     % If everything is ok this function is silent and changes nothing.
     VIPs0 = vips_port_mappings(VIPs),
     VIPs1 = healthy_vips(VIPs0, Nodes, Tree),
     VIPsP = prepare_vips(VIPs1),
-    Routes = get_vip_routes(VIPs1),
-    lists:foreach(fun (Namespace) ->
-        NamespaceBin = namespace2bin(Namespace),
-        LogPrefix = <<"netns: ", NamespaceBin/binary, "; ">>,
+    Routes = get_vip_routes(VIPs),
+    Diffs =
+        lists:map(fun (Namespace) ->
+            NamespaceBin = namespace2bin(Namespace),
+            LogPrefix = <<"netns: ", NamespaceBin/binary, "; ">>,
+
+            PrevRoutes = get_routes(RouteMgr, Namespace),
+            DiffRoutes = dcos_net_utils:complement(Routes, PrevRoutes),
+
+            PrevVIPsP = get_vips(IPVSMgr, Namespace),
+            DiffVIPs = diff(PrevVIPsP, VIPsP),
 
-        PrevRoutes = get_routes(RouteMgr, Namespace),
-        {RoutesToAdd, RoutesToDel} =
-            dcos_net_utils:complement(Routes, PrevRoutes),
+            {Namespace, LogPrefix, DiffRoutes, DiffVIPs}
+        end, Namespaces),
```

Comment:
Have you thought about having metrics for ipset? If you have some, maybe we can add a ticket to add those metrics part of the l4lb epic?

Reply:
Oh. O just found that on the spreadsheet. Nice :).